sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java

```diff
@@ -58,76 +60,255 @@ public static TraceConfig getDefault() {
+      int maxNumberOfEvents,
+      int maxNumberOfLinks,
+      int maxNumberOfAttributesPerEvent,
+      int maxNumberOfAttributesPerLink) {
+    this.sampler = sampler;
+    this.maxNumberOfAttributes = maxNumberOfAttributes;
+    this.maxNumberOfEvents = maxNumberOfEvents;
+    this.maxNumberOfLinks = maxNumberOfLinks;
+    this.maxNumberOfAttributesPerEvent = maxNumberOfAttributesPerEvent;
+    this.maxNumberOfAttributesPerLink = maxNumberOfAttributesPerLink;
+  }
 
   /**
    * Returns the global default {@code Sampler} which is used when constructing a new {@code Span}.
    *
    * @return the global default {@code Sampler}.
    */
-  public abstract Sampler getSampler();
+  public Sampler getSampler() {
+    return sampler;
+  }
 
   /**
    * Returns the global default max number of attributes per {@link Span}.
    *
    * @return the global default max number of attributes per {@link Span}.
    */
-  public abstract int getMaxNumberOfAttributes();
+  public int getMaxNumberOfAttributes() {
+    return maxNumberOfAttributes;
+  }
 
   /**
    * Returns the global default max number of {@link Event}s per {@link Span}.
    *
    * @return the global default max number of {@code Event}s per {@code Span}.
    */
-  public abstract int getMaxNumberOfEvents();
+  public int getMaxNumberOfEvents() {
+    return maxNumberOfEvents;
+  }
 
   /**
    * Returns the global default max number of {@link Link} entries per {@link Span}.
    *
    * @return the global default max number of {@code Link} entries per {@code Span}.
    */
-  public abstract int getMaxNumberOfLinks();
+  public int getMaxNumberOfLinks() {
+    return maxNumberOfLinks;
+  }
 
   /**
    * Returns the global default max number of attributes per {@link Event}.
    *
    * @return the global default max number of attributes per {@link Event}.
    */
-  public abstract int getMaxNumberOfAttributesPerEvent();
+  public int getMaxNumberOfAttributesPerEvent() {
+    return maxNumberOfAttributesPerEvent;
+  }
 
   /**
    * Returns the global default max number of attributes per {@link Link}.
    *
    * @return the global default max number of attributes per {@link Link}.
    */
-  public abstract int getMaxNumberOfAttributesPerLink();
+  public int getMaxNumberOfAttributesPerLink() {
+    return maxNumberOfAttributesPerLink;
+  }
 
   /**
    * Returns a new {@link Builder}.
    *
    * @return a new {@link Builder}.
    */
-  private static Builder newBuilder() {
-    return new AutoValue_TraceConfig.Builder();
+  public static Builder newBuilder() {
+    return new Builder();
   }
 
   /**
    * Returns a {@link Builder} initialized to the same property values as the current instance.
    *
    * @return a {@link Builder} initialized to the same property values as the current instance.
    */
-  public abstract Builder toBuilder();
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  public static final class Builder extends ConfigBuilder<Builder> {
```

Comment:
Also here, please use AutoValue. A reference of how using AutoValue with ConfigBuilder is available [here](https://github.com/dynatrace-oss-contrib/opentelemetry-java/blob/config-builder/exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinExporterConfiguration.java#L80)